luminex/src/org/labkey/luminex/LuminexExcelParser.java

```diff
@@ -242,6 +253,93 @@ private void parseFile() throws ExperimentException
         _parsed = true;
     }
 
+    // Keeps track of potential titrations using the following structure. 'count_descriptionM_analyteN' indicates the number of instances where descriptionM and analyteN are found on the same row.
+    // crossFilePT: {description1: {analyte1: count_description1_analyte1, analtye2: count_description1_analyte2, ...},
+    //               description2: {analyte1: count_description2_analyte1, analyte2: count_description2_analyte2, ...} ... }
+    private Map<String, HashMap<String, Integer>> buildCrossFilePTs(Map<String, HashMap<String, Integer>> crossFilePT, Map<String, Titration> potentialTitrations, String analyteName, Map<String, Integer> potentialTitrationCounts, String type)
+    {
+        for (String desc : potentialTitrations.keySet())
+        {
+            if (crossFilePT.containsKey(desc))
+            {
+                if (crossFilePT.get(desc).containsKey(analyteName))
+                {
+                    Integer count = crossFilePT.get(desc).get(analyteName);
+                    crossFilePT.get(desc).put(analyteName, count + potentialTitrationCounts.get(desc));
+                }
+                else
+                {
+                    crossFilePT.get(desc).put(analyteName, potentialTitrationCounts.get(desc));
+                }
+            }
+            else
+            {
+                HashMap<String, Integer> mini = new HashMap<>();
+
+                // Check for single-file titrations. Does not check for dilution counts.

```

Comment:
I would probably factor out this single-file titration check and call it above where you call `buildCrossFilePTs()`. Then this method doesn't have two functions, there would be one for building the cross file PTs and one for checking the single file PTs.

Reply:
Done!